src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Tests.cs

```diff
@@ -253,5 +255,56 @@ public void SkipRecursionIntoBaseDirectorySymlink()
 
             Assert.Null(reader.GetNextEntry());
         }
+
+        [ConditionalTheory(nameof(CanExtractWithTarTool))]
+        [MemberData(nameof(GetPaxAndGnuTestCaseNames))]
+        public void ResultingArchive_CanBeExtractedByOtherTools(string testCaseName)
+        {
+            if (testCaseName == "specialfiles" && !PlatformDetection.IsSuperUser)
+            {
+                throw new SkipTestException("specialfiles is only supported on Unix and it needs sudo permissions for extraction");
+            }
+
+            using TempDirectory root = new TempDirectory();
+
+            string archivePath = GetTarFilePath(CompressionMethod.Uncompressed, TestTarFormat.pax.ToString(), testCaseName);
+            string tarExtractedPath = Path.Join(root.Path, "tarExtracted");
+            Directory.CreateDirectory(tarExtractedPath);
+
+            // extract with /usr/bin/tar
+            ExtractWithTarTool(archivePath, tarExtractedPath);
+
+            // create archive with TarFile
+            string dotnetTarArchive = Path.Join(root.Path, "dotnetTarArchive.tar");
+            TarFile.CreateFromDirectory(tarExtractedPath, dotnetTarArchive, includeBaseDirectory: false);
+
+            // extract TarFile's archive with /usr/bin/tar
+            string dotnetTarExtractedPath = Path.Join(root.Path, "dotnetTarExtracted");
+            Directory.CreateDirectory(dotnetTarExtractedPath);
+            ExtractWithTarTool(dotnetTarArchive, dotnetTarExtractedPath);
+
+            // compare results
+            Dictionary<string, FileSystemInfo> expectedEntries = GetFileSystemInfosRecursive(tarExtractedPath).ToDictionary(fsi => fsi.FullName.Substring(tarExtractedPath.Length));
+
+            foreach (var actualEntry in GetFileSystemInfosRecursive(dotnetTarExtractedPath))
+            {
+                string keyPath = actualEntry.FullName.Substring(dotnetTarExtractedPath.Length);
+
+                Assert.True(expectedEntries.TryGetValue(keyPath, out FileSystemInfo expectedEntry), "Entry not expected.");
+                Assert.Equal(expectedEntry.Attributes, actualEntry.Attributes);
+                Assert.Equal(expectedEntry.LinkTarget, actualEntry.LinkTarget);
+
+                expectedEntries.Remove(keyPath);
+            }
+
+            // All expected entries should've been found and removed.
+            Assert.Equal(0, expectedEntries.Count);
+        }
+
+        private static FileSystemEnumerable<FileSystemInfo> GetFileSystemInfosRecursive(string path)
```

Comment:
@Jozkee can you please move this to TarTestBase.cs?We can reuse the method in https://github.com/dotnet/runtime/issues/75215